cardano-node/src/Cardano/Node/Types.hs

```diff
@@ -45,13 +59,100 @@ import           Ouroboros.Network.Block (MaxSlotNo (..))
+  { unDB :: FilePath }
+  deriving newtype Show
+
+newtype GenesisFile = GenesisFile
+  { unGenesisFile :: FilePath }
+  deriving stock (Eq, Ord)
+  deriving newtype (IsString, Show)
+
+instance FromJSON GenesisFile where
+  parseJSON (String genFp) = pure . GenesisFile $ Text.unpack genFp
+  parseJSON invalid = panic $ "Parsing of GenesisFile failed due to type mismatch. "
+                           <> "Encountered: " <> (Text.pack $ show invalid)
+
+-- Node can be run in two modes.
+data ViewMode = LiveView    -- Live mode with TUI
+              | SimpleView  -- Simple mode, just output text.
+              deriving (Eq, Show)
+
+instance FromJSON ViewMode where
+  parseJSON (String str) = case str of
+                            "LiveView" -> pure LiveView
+                            "SimpleView" -> pure SimpleView
+                            view -> panic $ "Parsing of ViewMode: "
+                                          <> view <> " failed. "
+                                          <> view <> " is not a valid view mode"
+  parseJSON invalid = panic $ "Parsing of ViewMode failed due to type mismatch. "
+                            <> "Encountered: " <> (Text.pack $ show invalid)
+
+newtype MaxConcurrencyBulkSync = MaxConcurrencyBulkSync
+  { unMaxConcurrencyBulkSync :: Word }
+  deriving stock (Eq, Ord)
+  deriving newtype (FromJSON, Show)
+
+newtype MaxConcurrencyDeadline = MaxConcurrencyDeadline
+  { unMaxConcurrencyDeadline :: Word }
+  deriving stock (Eq, Ord)
+  deriving newtype (FromJSON, Show)
+
+
+-- | IPv4 address with a port number.
+data NodeAddress = NodeAddress
+  { naHostAddress :: !NodeHostAddress
+  , naPort :: !PortNumber
+  } deriving (Eq, Ord, Show)
+
+instance FromJSON NodeAddress where
+  parseJSON = withObject "NodeAddress" $ \v -> do
+    NodeAddress
+      <$> v .: "addr"
+      <*> ((fromIntegral :: Int -> PortNumber) <$> v .: "port")
+
+instance ToJSON NodeAddress where
+  toJSON na =
+    object
+      [ "addr" .= toJSON (naHostAddress na)
+      , "port" .= (fromIntegral (naPort na) :: Int)
+      ]
+
+-- Embedding a Maybe inside a newtype is somewhat icky but this seems to work
+-- and removing the Maybe breaks the functionality in a subtle way that is difficult
+-- to diagnose.
+newtype NodeHostAddress
+  = NodeHostAddress { unNodeHostAddress :: Maybe IP }
+  deriving newtype Show
+  deriving (Eq, Ord)
+
+instance FromJSON NodeHostAddress where
+  parseJSON (String ipStr) =
+    case readMaybe $ Text.unpack ipStr of
+      Just ip -> pure $ NodeHostAddress (Just ip)
+      Nothing -> panic $ "Parsing of IP failed: " <> ipStr
+  parseJSON Null = pure $ NodeHostAddress Nothing
+  parseJSON invalid = panic $ "Parsing of IP failed due to type mismatch. "
+                            <> "Encountered: " <> (Text.pack $ show invalid) <> "\n"
+
+instance ToJSON NodeHostAddress where
+  toJSON mha =
+    case unNodeHostAddress mha of
+      Just ip -> String (Text.pack $ show ip)
+      Nothing -> Null
```

Comment:
Note that all of this is duplicated in `Cardano.CLI.Types`. Is that intentional?

Reply:
Yep at this point it is. I've just realized some of these types are tied to "dead" cli commands.

Reply:
Removed along with the cli commands that were using them